File: C/exercicios1/5.c
#include <stdio.h>
#include <string.h>

int contarOcorrencias(char *frase, char *palavra) {
    int ocorrencias = 0;
    int tamanhoFrase = strlen(frase);
    int tamanhoPalavra = strlen(palavra);

    for (int i = 0; i <= tamanhoFrase - tamanhoPalavra; i++) {
        if (strncmp(&frase[i], palavra, tamanhoPalavra) == 0) {
            ocorrencias++;
        }
    }

    return ocorrencias;
}

int main() {
    char frase[100];
    char palavra[20];

    printf("Digite a frase: ");
    fgets(frase, 100, stdin);
    frase[strcspn(frase, "\n")] = '\0';

    printf("Digite a palavra: ");
    scanf("%s", palavra);

    int ocorrencias = contarOcorrencias(frase, palavra);

    printf("Temos que a palavra %s ocorre %d vezes na frase.\n", palavra, ocorrencias);

    return 0;
}

File: Go/Exc1/5.go
package main

import "fmt"
import "bufio"
import "os"
import "strings"

func main () {
	reader := bufio.NewReader(os.Stdin)
	
	fmt.Print("Informe a frase: ")
	frase, _ := reader.ReadString('\n')
	frase = strings.TrimSpace(frase)

	fmt.Print("Informe a palavra: ")
	palavra, _ := reader.ReadString('\n')
	palavra = strings.TrimSpace(palavra)
	
	ocorrencias := strings.Count(frase, palavra)
	
	fmt.Printf("A palavra '%s' ocorre %d vezes na frase.\n", palavra, ocorrencias)
}
